src/Model/Item.php

```diff
@@ -2271,6 +2271,11 @@ private static function autoReshare(array $item)
 			return;
 		}
 
+		$cdata = Contact::getPublicAndUserContactID($item['author-id'], $item['uid']);
+		if (empty($cdata['user']) || ($cdata['user'] != $item['contact-id'])) {
```

Comment:
I thought `item.contact-id`, `item.author-id` and `item.owner-id` were all public contact ids (with `uid = 0`)?

Reply:
`contact-id` mostly is the id of the responsible contact in the user's context. It can be the same like the `author-id` but normally it isn't. That field is some kind of relict of the old datanase structure.